File: core/src/models/organization.rs
use chrono::{DateTime, Utc};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, FromRow)]
pub struct Organization {
    pub id: Uuid,
    pub name: String,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone)]
pub enum OrganizationRole {
    Admin,
    Member,
}

#[derive(Debug, Clone, FromRow)]
pub struct OrganizationMember {
    pub id: Uuid,
    pub user_id: Uuid,
    pub organization_id: Uuid,
    pub role: OrganizationRole,
    pub created_at: DateTime<Utc>,
}

File: gitdot-api-ts/src/resource/organization.ts
import { z } from "zod";

export const OrganizationResource = z.object({
  id: z.string().uuid(),
  name: z.string(),
  created_at: z.string().datetime(),
});
export type OrganizationResource = z.infer<typeof OrganizationResource>;

export const OrganizationMemberResource = z.object({
  id: z.string().uuid(),
  user_id: z.string().uuid(),
  organization_id: z.string().uuid(),
  role: z.string(),
  created_at: z.string().datetime(),
});
export type OrganizationMemberResource = z.infer<
  typeof OrganizationMemberResource
>;
